pilot/helpers/Debugger.py

```diff
@@ -22,15 +22,26 @@ def debug(self, convo, command=None, user_input=None, issue_description=None, is
             command (dict, optional): The command to debug. Default is None.
             user_input (str, optional): User input for debugging. Default is None.
             issue_description (str, optional): Description of the issue to debug. Default is None.
+            is_root_task (bool, optional): Pass to `developer.execute_task`
 
         Returns:
             bool: True if debugging was successful, False otherwise.
         """
 
         self.recursion_layer += 1
         if self.recursion_layer > MAX_RECUSION_LAYER:
+            # Provide detailed error information
+            details = {
+                "Current Recursion Layer": self.recursion_layer,
+                "Max Allowed Recursion Layer": MAX_RECUSION_LAYER,
+                "Conversation": str(convo),
+                "Command": command,
+                "User Input": user_input,
+                "Issue Description": issue_description
+            }
+            error_message = "\n".join([f"{key}: {value}" for key, value in details.items()])
             self.recursion_layer = 0
-            raise TooDeepRecursionError()
+            raise TooDeepRecursionError(context=error_message)
```

Comment:
Isn't this going to show all level errors?
For example if we go 3 levels deep, this will at the end show all 3 levels of errors, right?

What we want to do is show only initial error

Reply:
I believe this will only show fourth error.
As I understand, i take that as `TooDeepRecursionError` will be raised at recursion_layer = 4, like previous handling. It will not raised when recursion_layer = 3.
I believe i haven't changed the origin logic but just add detailed info when it once shows up.
```
if self.recursion_layer > MAX_RECUSION_LAYER:
            # Provide detailed error information
            details = {
                "Current Recursion Layer": self.recursion_layer,
                "Max Allowed Recursion Layer": MAX_RECUSION_LAYER,
                "Conversation": str(convo),
                "Command": command,
                "User Input": user_input,
                "Issue Description": issue_description
            }
            error_message = "\n".join([f"{key}: {value}" for key, value in details.items()])
            self.recursion_layer = 0
            raise TooDeepRecursionError(context=error_message)
```